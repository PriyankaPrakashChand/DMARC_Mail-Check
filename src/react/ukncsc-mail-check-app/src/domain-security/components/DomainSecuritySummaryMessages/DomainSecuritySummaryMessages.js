import React from 'react';
import PropTypes from 'prop-types';
import { MailCheckMessage } from 'common/components';

const DomainSecuritySummaryMessages = ({
  type,
  failures,
  warnings,
  inconclusives,
}) => (
  <React.Fragment>
    {!!failures && (
      <MailCheckMessage error>
        {type} configuration has {failures} urgent issue{failures === 1
          ? ''
          : 's'}
      </MailCheckMessage>
    )}
    {!!warnings && (
      <MailCheckMessage warning>
        {type} configuration has {warnings} advisory issue{warnings === 1
          ? ''
          : 's'}
      </MailCheckMessage>
    )}
    {!!inconclusives && (
      <MailCheckMessage>
        {type} configuration has {inconclusives} inconclusive issue{inconclusives ===
        1
          ? ''
          : 's'}
      </MailCheckMessage>
    )}
    {!failures &&
      !warnings &&
      !inconclusives && (
        <MailCheckMessage success>{type} well configured.</MailCheckMessage>
      )}
  </React.Fragment>
);

DomainSecuritySummaryMessages.defaultProps = {
  failures: 0,
  warnings: 0,
  inconclusives: 0,
};

DomainSecuritySummaryMessages.propTypes = {
  type: PropTypes.string.isRequired,
  failures: PropTypes.number,
  warnings: PropTypes.number,
  inconclusives: PropTypes.number,
};

export default DomainSecuritySummaryMessages;
