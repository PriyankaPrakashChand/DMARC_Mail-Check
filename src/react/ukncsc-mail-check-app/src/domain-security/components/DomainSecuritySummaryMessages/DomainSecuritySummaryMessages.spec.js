import React from 'react';
import { render } from 'react-testing-library';
import DomainSecuritySummaryMessages from './DomainSecuritySummaryMessages';

describe('DomainSecuritySummaryMessages', () => {
  let container;

  describe('when failures or warnings are present', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecuritySummaryMessages
          type="DMARC"
          title="bleh"
          failures={1}
          warnings={2}
        />
      ));
    });

    it('should match the snapshot', () => expect(container).toMatchSnapshot());

    it('should render the error mesage', () =>
      expect(container.getElementsByClassName('ui error message')).toHaveLength(
        1
      ));

    it('should render the warning message', () =>
      expect(
        container.getElementsByClassName('ui warning message')
      ).toHaveLength(1));

    it('should not render a success message', () =>
      expect(
        container.getElementsByClassName('ui success message')
      ).toHaveLength(0));
  });

  describe('when no failures or warnings are present', () => {
    beforeEach(() => {
      ({ container } = render(
        <DomainSecuritySummaryMessages
          type="SPF"
          title="bleh"
          failures={0}
          warnings={0}
        />
      ));
    });

    it('should render a success message', () =>
      expect(
        container.getElementsByClassName('ui success message')
      ).toHaveLength(1));
  });
});
