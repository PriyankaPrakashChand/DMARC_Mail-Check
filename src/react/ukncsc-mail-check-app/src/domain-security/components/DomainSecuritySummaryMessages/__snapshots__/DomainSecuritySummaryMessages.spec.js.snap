// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DomainSecuritySummaryMessages when failures or warnings are present should match the snapshot 1`] = `
<div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui error message MailCheckMessage--container"
  >
    DMARC
    configuration has 
    1
    urgent issue
  </div>
  <div
    style="margin: 0.2em 0em; max-width: 600px;"
    class="ui warning message MailCheckMessage--container"
  >
    DMARC
    configuration has 
    2
    advisory issue
    s
  </div>
</div>
`;
